Snowplow publishing doc (#704)
avohq/docs: pages/workspace/tracking-plan/publishing.mdx

  style={{ margin: '32px 0 -32px 0' }}
/>

## <a name="snowplow"></a> Snowplow Data Structures

### Configure the Snowplow integration

To enable the direct integration with Snowplow, you need to configure it with the following parameters.

- **Snowplow Organization ID**: Your organization ID issued by Snowplow.
- **Snowplow Vendor Name**: The vendor name of your Snowplow schemas.
- **Snowplow Access Token**: The secret access token to update your Snowplow data structures.

### Publishing to Snowplow

You can publish your Avo tracking plan into Snowplow with the publishing integration.

> > Data structures interfaces are only compatible with pipelines that have been upgraded to use Iglu Server registries, rather than static S3 registries. Please check here in the console, to see if you need an upgrade or if your registries are ready to go.

Reviewer: > Please check here in the console

What does this mean?

> > Data structures interfaces are only compatible with pipelines that have been upgraded to use Iglu Server registries, rather than static S3 registries. Please check here in the console, to see if you need an upgrade or if your registries are ready to go.

Author: ```suggestion
> Data structures interfaces are only compatible with pipelines that have been upgraded to use Iglu Server registries, rather than static S3 registries. Please check in your [Snowplow console](https://console.snowplowanalytics.com/data-structures/), to see if you need an upgrade or if your registries are ready to go.
```

  style={{ margin: '32px 0 -32px 0' }}
/>

## <a name="snowplow"></a> Snowplow Data Structures

### Configure the Snowplow integration

To enable the direct integration with Snowplow, you need to configure it with the following parameters.

- **Snowplow Organization ID**: Your organization ID issued by Snowplow.
- **Snowplow Vendor Name**: The vendor name of your Snowplow schemas.
- **Snowplow Access Token**: The secret access token to update your Snowplow data structures.

### Publishing to Snowplow

You can publish your Avo tracking plan into Snowplow with the publishing integration.

> > NOTE

Author: ```suggestion
```

> Data structures interfaces are only compatible with pipelines that have been upgraded to use Iglu Server registries, rather than static S3 registries. Please check here in the console, to see if you need an upgrade or if your registries are ready to go.

The Snowplow API enforces a workflow of validating, testing on development and then deploying to production.

Avo publishing integration will take care of validating and deploying your data structures to the development environment, where you take over to test the data structures and promote the tested data structures to production. 

> Note that only Admin users can promote data structures from Dev to Production. Learn more on how to promote the data structures with Snowplow Console (here)[https://docs.snowplow.io/docs/understanding-tracking-design/managing-data-structures/#promoting-a-data-structure]

Author: ```suggestion
Note that only Admin users can promote data structures from Dev to Production. Learn more on how to promote the data structures with Snowplow Console [here](https://docs.snowplow.io/docs/understanding-tracking-design/managing-data-structures/#promoting-a-data-structure)
```

### Versioning

Avo will create new data structures with version 1-0-0 and will patch the existing data structures in your development environment on the same version as they are.

> When promoting the tested data structures to production you will be required update the version. Read more about the versioning (here)[https://docs.snowplow.io/docs/understanding-tracking-design/versioning-your-data-structures/]

Author: ```suggestion
When promoting the tested data structures to production you will be required update the version. Read more about the versioning [here](https://docs.snowplow.io/docs/understanding-tracking-design/versioning-your-data-structures/)
```
